dotnet/src/Agents/Framework/ILocalAgent.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// Contract for an agent that acts upon a local message history.
+/// </summary>
+public interface ILocalAgent
```

Comment:
The name suggests this defines a Local Agent but this is an interface that defines the contract to invoke an Agent with a chant history.

Reply:
IAgentWithChatHistory? just thinking loud (now catching up with the ADR but will dig in deeper with the code...)
Question(s) to the masters:
- My understanding is that this PR implements https://github.com/microsoft/semantic-kernel/blob/adr_agent_architecture/docs/decisions/0039-agent-interactions.md and is aligned with https://github.com/microsoft/semantic-kernel/blob/adr_agent_architecture/docs/decisions/0032-agents.md - yes?
- What would be the best way to get the bits aside from getting this branch and building it locally? do you have a sample configuration file to fulfill? (that would help lots)

I will dig "quite a bit" in as I have a Session on the 19th April on Global Azure - on Agents ;)
So I plan to show some "what can we do now" and "in the short future" - again thanks for building this - it is amazing!!

Reply:
https://github.com/microsoft/semantic-kernel/issues/5780